server/src/controllers/Brutes.ts

```diff
@@ -1190,6 +1194,135 @@ const Brutes = {
+      if (!isWeapon && !isSkill && !isPet) {
+        throw new Error('Wrong choice type');
+      }
+
+      const ascendChoice = choice as WeaponName | SkillName | PetName;
+
+      const userBrute = await prisma.brute.findFirst({
+        where: {
+          name: ilike(name),
+          deletedAt: null,
+          userId: authed.id,
+        },
+        select: {
+          id: true,
+          ranking: true,
+          canRankUpSince: true,
+          destinyPath: true,
+          clanId: true,
+          level: true,
+          eventId: true,
+          xp: true,
+          weapons: true,
+          skills: true,
+          pets: true,
+          ascendedWeapons: true,
+          ascendedSkills: true,
+          ascendedPets: true,
+          destinyChoices: true,
+        },
+      });
+
+      if (!userBrute) {
+        throw new ExpectedError(translate('bruteNotFound', authed));
+      }
+
+      if (!userBrute.canRankUpSince) {
+        throw new ExpectedError(translate('bruteCannotRankUp', authed));
+      }
+
+      if (userBrute.ranking !== 0) {
+        throw new ExpectedError(translate('bruteNotMaxRank', authed));
+      }
+
+      if (isWeapon) {
+        if (!userBrute.weapons.includes(ascendChoice as WeaponName)) {
+          throw new ExpectedError(translate('bruteMissingWeapon', authed));
+        }
+        if (userBrute.ascendedWeapons.includes(ascendChoice as WeaponName)) {
+          throw new ExpectedError(translate('bruteWeaponAlreadyAscended', authed));
+        }
+      }
+      if (isSkill) {
+        if (!userBrute.skills.includes(ascendChoice as SkillName)) {
+          throw new ExpectedError(translate('bruteMissingSkill', authed));
+        }
+        if (userBrute.ascendedSkills.includes(ascendChoice as SkillName)) {
+          throw new ExpectedError(translate('bruteSkillAlreadyAscended', authed));
+        }
+      }
+      if (isPet) {
+        if (!userBrute.pets.includes(ascendChoice as PetName)) {
+          throw new ExpectedError(translate('bruteMissingPet', authed));
+        }
+        if (userBrute.ascendedPets.includes(ascendChoice as PetName)) {
+          throw new ExpectedError(translate('brutePetAlreadyAscended', authed));
+        }
+      }
+
+      const brute = await resetBrute({
+        prisma,
+        brute: userBrute,
+        free: true,
+        ascended: ascendChoice,
+      });
+
+      // Achievement
+      await increaseAchievement(prisma, authed.id, brute.id, 'ascend');
+
+      const ascensions = brute.ascendedSkills.length
+        + brute.ascendedWeapons.length
+        + brute.ascendedPets.length;
+
+      // Add ascend log
+      await prisma.log.create({
+        data: {
+          currentBruteId: brute.id,
+          type: LogType.ascend,
+          level: ascensions,
+        },
+        select: { id: true },
+      });
+
+      // Update clan points
+      if (userBrute.clanId) {
+        await updateClanPoints(prisma, userBrute.clanId, 'add', brute, userBrute);
+      }
```

Comment:
This shouldn't be needed as the brute isn't added to the clan, and its clan points doesn't change after ascending.

Reply:
doesn't the level change the amount of point ?

Reply:
Yep, that's my bad, I forgot about that !